### PR TITLE
Add webhook to void payments

### PR DESCRIPTION
**Pull request by Naokimi**
Fixes https://github.com/nebulab/solidus_bolt/issues/34

### CODE REVIEW

File: app/services/solidus_bolt/payments/void_sync_service.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+      def initialize(payment:)
+        @payment = payment
+        super
+      end
+
+      def call
+        return if payment.void?
+
+        payment.void!
```
Comment (piyushswain) [inline app/services/solidus_bolt/payments/void_sync_service.rb:16]: can we simplify this by writing it like

```rb
payment.void! unless payment.void?
```

File: spec/services/solidus_bolt/payments/void_sync_service_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+RSpec.describe SolidusBolt::Payments::VoidSyncService do
+  subject(:void) { described_class.call(payment: payment) }
+
+  let(:payment) { create(:bolt_payment) }
+
+  describe '#call' do
+    it 'voids the payment' do
+      expect { void }.to change { payment.reload.state }.from('checkout').to('void')
+    end
+  end
```
Comment (piyushswain) [inline spec/services/solidus_bolt/payments/void_sync_service_spec.rb:14]: Can we also test the situation where the payment is already in `void` state ?

**Review by piyushswain — CHANGES_REQUESTED** [review]
Looks mostly fine. Just a few minor comments

**Review by DanielePalombo — APPROVED** [review]
I agree with the @piyushswain 's comments. I think we can merge this PR after you will address them.

File: spec/services/solidus_bolt/payments/void_sync_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+require 'spec_helper'
+
+RSpec.describe SolidusBolt::Payments::VoidSyncService do
+  subject(:void) { described_class.call(payment: payment) }
+
+  let(:payment) { create(:bolt_payment) }
+
+  describe '#call' do
+    context 'when non-void payment' do
```
Comment (DanielePalombo) [inline spec/services/solidus_bolt/payments/void_sync_service_spec.rb:11]: ❤️

**Review by piyushswain — APPROVED** [review]
LGTM

**Pull request merged**